test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```diff
@@ -669,4 +673,175 @@ private void compare(final SortingRule got, final SortingRule expected) {
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(new LinkedHashSet(adminList))
+                        .withAdmin(true)
+                        .withPublic(false)))
+                .thenReturn(privateData);
+
+
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(set())
+                        .withAdmin(true)
+                        .withPublic(true)))
+                .thenReturn(publicData);
+
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(set())
+                        .withPublic(true)))
+                .thenReturn(publicData);
+
+        //should not happen
+        when(is.searchTypes( kbasesearchengine.search.MatchFilter.getBuilder().build(),
+                new kbasesearchengine.search.AccessFilter()
+                        .withAccessGroups(set())
+                        .withPublic(false)))
+                .thenReturn(new HashMap<>());
+        return sm;
+    }
+    @Test
+    public void testAccessFilterAuthorizedUser() throws Exception{
+        SearchInterface sm = setUpTestAccessFilter();
+        //expected private results: SearchMethods changes integer results to longs
+        final Map<String, Long> privateData = new HashMap<>();
+        privateData.put("test", 1L);
+
+        final Map<String, Long> allData = new HashMap<>();
+        allData.put("test", 1L);
+        allData.put("notdefault", 1L);
+
+        final Map<String, Long> publicData = new HashMap<>();
+        publicData.put("notdefault", 1L);
+
+
+        final SearchTypesOutput res = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+
+        assertThat("should show both public and private", res.getTypeToCount(), is(allData));
+
+
+        final SearchTypesOutput res2 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(0L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only public", res2.getTypeToCount(), is(publicData));
+
+        final SearchTypesOutput res3 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(0L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only private", res3.getTypeToCount(), is(privateData));
+
+
+        final SearchTypesOutput res4 = sm.searchTypes(new SearchTypesInput()
```

Comment:
This is a weird case, as mentioned previously. I sort of think it should throw an exception because it's guaranteed to return no data, so what's the point? 

That can wait for another PR if we go that way though.

Reply:
so remove this and add the mockito verify to the top condition

Reply:
so if I am throwing an exception in the above stubbed case, should I have this in a try catch and see if it got the right exception or just remove this case

Reply:
I'm not sure - I assumed the stubbed case above was impossible. If it's possible that case could occur (or will occur in your tests) then throwing an exception isn't the right thing to do. 

IIUC, the stubbed case is when a user essentially makes a call that says "don't do a search", right? If so, that's currently allowed by the api, and I think goes to elasticsearch which will return, effectively, an empty list.

Ugh, you know what, it'd be way simpler if we just made the method throw an IllegalArgumentException if public and private are both false and make a quick test case for it. Then we don't have to think about this, assuming I'm understanding the issue.